Source/CesiumRuntime/Private/CesiumGeoJsonObject.cpp

```diff
@@ -0,0 +1,412 @@
+#include "CesiumGeoJsonObject.h"
+
+#include "CesiumGeospatial/Cartographic.h"
+#include "Dom/JsonObject.h"
+
+#include <utility>
+#include <variant>
+#include <vector>
+
+int64 UCesiumGeoJsonFeatureBlueprintLibrary::GetIdAsInteger(
+    const FCesiumGeoJsonFeature& InFeature) {
+  if (!InFeature._document || !InFeature._feature) {
+    return -1;
+  }
+  const int64_t* pId = std::get_if<int64_t>(&InFeature._feature->id);
+  if (!pId) {
+    return -1;
+  }
+
+  return *pId;
+}
+
+FString UCesiumGeoJsonFeatureBlueprintLibrary::GetIdAsString(
+    const FCesiumGeoJsonFeature& InFeature) {
+  if (!InFeature._document || !InFeature._feature) {
+    return {};
+  }
+  struct GetIdVisitor {
+    FString operator()(const int64_t& id) { return FString::FromInt(id); }
+    FString operator()(const std::string& id) {
+      return UTF8_TO_TCHAR(id.c_str());
+    }
+    FString operator()(const std::monostate& id) { return {}; }
+  };
+
+  return std::visit(GetIdVisitor{}, InFeature._feature->id);
+}
+
+namespace {
+TSharedPtr<FJsonValue>
+jsonValueToUnrealJsonValue(const CesiumUtility::JsonValue& value) {
+  struct JsonValueVisitor {
+    TSharedPtr<FJsonValue> operator()(const CesiumUtility::JsonValue::Null&) {
+      return MakeShared<FJsonValueNull>();
+    }
+    TSharedPtr<FJsonValue>
+    operator()(const CesiumUtility::JsonValue::Bool& value) {
+      return MakeShared<FJsonValueBoolean>(value);
+    }
+    TSharedPtr<FJsonValue> operator()(const std::string& value) {
+      return MakeShared<FJsonValueString>(UTF8_TO_TCHAR(value.c_str()));
+    }
+    TSharedPtr<FJsonValue> operator()(const double& value) {
+      return MakeShared<FJsonValueNumber>(value);
+    }
+    TSharedPtr<FJsonValue> operator()(const std::uint64_t& value) {
+      return MakeShared<FJsonValueNumberString>(FString::FromInt(value));
+    }
+    TSharedPtr<FJsonValue> operator()(const std::int64_t& value) {
+      return MakeShared<FJsonValueNumberString>(FString::FromInt(value));
+    }
+    TSharedPtr<FJsonValue>
+    operator()(const CesiumUtility::JsonValue::Array& value) {
+      TArray<TSharedPtr<FJsonValue>> values;
+      values.Reserve(value.size());
+      for (const CesiumUtility::JsonValue& v : value) {
+        values.Emplace(jsonValueToUnrealJsonValue(v));
+      }
+      return MakeShared<FJsonValueArray>(MoveTemp(values));
+    }
+    TSharedPtr<FJsonValue>
+    operator()(const CesiumUtility::JsonValue::Object& value) {
+      TSharedPtr<FJsonObject> obj = MakeShared<FJsonObject>();
+      for (const auto& [k, v] : value) {
+        obj->SetField(UTF8_TO_TCHAR(k.c_str()), jsonValueToUnrealJsonValue(v));
+      }
+      return MakeShared<FJsonValueObject>(MoveTemp(obj));
```

Comment:
This looks redundant.

```suggestion
      return obj;
```

Reply:
For whatever reason, `FJsonValueObject` and `FJsonObject` are distinct types and we seem to need to manually convert one to the other like this.